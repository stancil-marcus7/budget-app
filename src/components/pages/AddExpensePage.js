import React from 'react';
import ExpenseForm from '../ExpenseForm'
import {connect} from 'react-redux' 
import {startAddExpense} from '../../actions/expenses'
export class AddExpensePage extends React.Component{
  onSubmit = (expense) => {
    this.props.startAddExpense(expense);
    //This will switch us over to the home page
    this.props.history.push('/');
  };

  render() {
    return (
      <div>
          <div className="page-header">
            <div className="content-container">
              <h1 className="page-header__title">Add Expense</h1>
            </div>
          </div>
          <div className="content-container">
            <ExpenseForm
                  button="Add Expense"
                  onSubmit={this.onSubmit}/>
          </div>
      </div>
    )
  }
}

//Allows you to create props out of dispatch
const mapDispatchToProps = (dispatch) => ({
    startAddExpense: (expense) => dispatch(startAddExpense(expense))
})

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
