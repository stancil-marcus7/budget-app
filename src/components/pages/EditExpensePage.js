import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from '../ExpenseForm'
import { editExpense } from '../../actions/expenses'
import { removeExpense } from '../../actions/expenses'

export class EditExpensePage extends React.Component {
render(){
  return (
    <div>
      <ExpenseForm
        expense={this.props.expense}
        onSubmit={expense => {
          this.props.editExpense(this.props.expense.id, expense);
          this.props.history.push('/');
        }}
        />
        <button onClick={expense => {
            this.props.removeExpense({id: this.props.expense.id})
            this.props.history.push('/')
          }}
            
        >Remove</button>
    </div>
  )
} 
}

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find(expense => expense.id === props.match.params.id
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  //need all params from action
  editExpense: (id, expense) => dispatch(editExpense(id, expense)),
  removeExpense: (expense) => dispatch(removeExpense({id: expense.id}))
})
export default connect(mapStateToProps,mapDispatchToProps)(EditExpensePage);
