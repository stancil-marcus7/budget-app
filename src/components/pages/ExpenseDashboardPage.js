import React from 'react';
import getExpenseTotal from "../../expenses-total";
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../../selectors/expenses'

export class ExpenseDashboardPage extends React.Component{
    render(){
        return (
            <div className="page-header">
                <div className="content-container">
                    <h1 className="page-header__title">Viewing <span>{this.props.expenseCount}</span> {this.props.expenseCount > 1 || this.props.expenseCount === 0 ? "expenses " : "expense "} totalling <span>{numeral(this.props.expenseTotal/100).format('$0,0.00')}</span></h1>
                    <div className="page-header__actions">
                        <Link className="button" to="/create">Add Expense</Link>
                    </div>
                </div>
                
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        expenseCount: selectExpenses(state.expenses, state.filters).length,
        expenseTotal: getExpenseTotal(selectExpenses(state.expenses, state.filters))
    }
}



export default connect(mapStateToProps)(ExpenseDashboardPage); 