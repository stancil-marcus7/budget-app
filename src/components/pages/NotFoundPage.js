import React from 'react';
import {NavLink} from 'react-router-dom'

const NotFoundPage = () => (
  <div>
      404! - <NavLink to="/">Go Home!</NavLink>
  </div>
)

export default NotFoundPage;
