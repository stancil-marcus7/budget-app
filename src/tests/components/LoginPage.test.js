import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/pages/LoginPage'

test(`should render LoginPage correctly`, () => {
    const wrapper = shallow(<LoginPage/>);
    expect(wrapper).toMatchSnapshot();
});

test(`should call startLogin on button click`, ()=> {
    let startLogin = jest.fn();  
    const wrapper = shallow(<LoginPage startLogin={startLogin}/>)
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled();
})