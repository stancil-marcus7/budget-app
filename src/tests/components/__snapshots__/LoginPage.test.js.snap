// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1>
      Expensify App
    </h1>
    <p>
      It's time to get your expenses under control
    </p>
    <button>
      Login
    </button>
  </div>
</div>
`;
